### PR TITLE
Make sure we can always find the right file

### PR DESCRIPTION
**Pull request by danpalmer**
Looks like there was an edge case in how find was finding particular files. This fixes it.

Would appreciate review from @justinabrahms in case you can think of any knock-on issues this might cause.

### CODE REVIEW

**Comment by justinabrahms** [comment]
Can you talk a little more about what the problem is? The second argument to find should be the directory to look in.

**Comment by danpalmer** [comment]
Here's an example. The repo is "thread/styleme", and imhotep is given `/cache` as the cache directory. 

```
# From the root, searching does not find the file
/# find /cache/thread__styleme -samefile "frontend/components.js"
No such file: frontend/components.js

# From the correct directory, searching does find the file with the full path
/cache/thread__styleme# find /cache/thread__styleme -samefile "frontend/components.js"
/cache/thread__styleme/frontend/components.js
```

I'm not sure this is a great fix – I've done it and switched our imhotep setup to use this, but only as a temporary measure until we can hopefully find a better solution.

**Comment by justinabrahms** [comment]
How about using `-path` instead?

```
$ mkdir -p /tmp/cache/thread__styleme/frontend
$ touch /tmp/cache/thread__styleme/frontend/components.js
$ tree /tmp/cache 
/tmp/cache
└── thread__styleme
    └── frontend
        └── components.js

2 directories, 1 file
$ find /tmp/cache/thread__styleme -samefile "frontend/components.js" 
find: ‘frontend/components.js’: No such file or directory
$ find /tmp/cache/thread__styleme  -name "frontend/components.js"
find: warning: Unix filenames usually don't contain slashes (though pathnames do).  That means that '-name ‘frontend/components.js’' will probably evaluate to false all the time on this system.  You might find the '-wholename' test more useful, or perhaps '-samefile'.  Alternatively, if you are using GNU grep, you could use 'find ... -print0 | grep -FzZ ‘frontend/components.js’'.
$ find /tmp/cache/thread__styleme -path "*/frontend/components.js"
/tmp/cache/thread__styleme/frontend/components.js
```

**Comment by gordol** [comment]
any plans to actually fix this?

pylint plugin doesn't work because of this.

**Comment by gordol** [comment]
if i make any changes to fix this, then it breaks all the other plugins...

File: imhotep/tools.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def invoke(self, dirname, filenames=set(), linter_configs=set()):
             to_find = ' -o '.join(['-name "*%s"' % ext
                                    for ext in self.get_file_extensions()])
 
-        cmd = 'find %s %s | xargs %s' % (
+        cmd = 'find %s -path "*/%s" | xargs %s' % (
```
Comment (gordol) [inline imhotep/tools.py]: this is making invalid find commands...

Comment (danpalmer) [reply imhotep/tools.py]: @gordol sorry for the late reply, I'm lacking some context here but looking to merge this if I can. Can you remember what was invalid about the commands it's generating here?

**Pull request merged**